include/daqdataformats/TriggerRecordHeader.hpp

```diff
@@ -326,6 +334,18 @@ TriggerRecordHeader::operator[](size_t idx)
   return *(reinterpret_cast<ComponentRequest*>(header_() + 1) + idx); // NOLINT
 }
 
+ComponentRequest const&
+TriggerRecordHeader::get_component_for_source_id(SourceID const& source_id) const
+{
+  for (uint64_t idx = 0; idx < get_num_requested_components(); ++idx) {
+    ComponentRequest const& component_obj = *(reinterpret_cast<ComponentRequest*>(header_() + 1) + idx); // NOLINT
+    if (source_id == component_obj.component) {
+      return component_obj;
+    }
+  }
+  throw std::invalid_argument("Supplied SourceID was not found in the ComponentRequest list.");
```

Comment:
Is throwing an exception the best thing to do here? What if the method returns `std::optional<ComponentRequest>` so it can be queried if it was found? (I don't know how well that works with Python bindings, so the exception may be better if this method is primarily intended for that use-case)